### ISSUE TITLE
Haml 6 not throwing Haml::HamlSyntaxError on syntax error

### ISSUE THREAD

**Issue by navidemad** [haml/haml#1082]
Hello,

When there is a syntax error in the dom, the parser returned different output since version 6.0.0.beta.1
Is it something done on purpose ?

```ruby
require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"
  gem "rails"

  # gem "haml", "6.0.0.beta.1"
  # FAILURE: 
  # returned the class Haml::HamlSyntaxError instead of raising Haml::HamlSyntaxError (inherited from Haml::HamlError)

  gem "haml", "5.2.2"
  # SUCCESS:
  # raised Haml::SyntaxError (inherited from Haml::Error)
end

require "minitest/autorun"

class BugTest < Minitest::Test
  # def test_haml_6_parser
  #   assert_raises Haml::HamlError do
  #     ret = Haml::Parser.new({}).call("%body\n  %div\n        %p")
  #     puts ret.class
  #     puts ret
  #     ret
  #   end
  # end

  def test_haml_5_parser
    assert_raises Haml::Error do
      ret = Haml::Parser.new({}).call("%body\n  %div\n        %p")
      puts ret.class
      puts ret
      ret
    end
  end
end
```

Thanks

**Comment by k0kubun** [comment]
Thank you! This reminded me of a forgotten issue that I should have worked on. I addressed this in v6.0.0.beta.2, so please give it a try.

**Comment by navidemad** [comment]
Such a quick fix @k0kubun, thanks and it's working with 6.0.0.beta.2 🥇
